SygicMapsKit/Classes/SearchModule/SYMKSearchViewController.swift

```diff
@@ -0,0 +1,183 @@
+    
+    /// Delegate receives information, that user did cancel action.
+    ///
+    /// - Parameter searchController: Search module controller.
+    func searchControllerDidCancel(_ searchController: SYMKSearchViewController)
+    
+    
+    /// Delegate event for error state handling
+    ///
+    /// - Parameters:
+    ///   - searchController: Search module controller.
+    ///   - searchState: state to handle
+    /// - Returns: String message to display for received search state. Return nil if you don't what to display any message.
+    func searchController(_ searchController: SYMKSearchViewController, willShowMessageFor searchState: SYRequestResultState) -> String?
+}
+
+public extension SYMKSearchViewControllerDelegate {
+    func searchController(_ searchController: SYMKSearchViewController, willShowMessageFor searchState: SYRequestResultState) -> String? {
+        return searchState.stringMessage()
+    }
+}
+
+
+/// Search module.
+///
+/// A controller that manages searching for point of interests, cities, streets etc.
+public class SYMKSearchViewController: SYMKModuleViewController {
+    
+    // MARK: - Public properties
+    
+    /// Results controller manages searched results.
+    public var resultsViewController: SYUISearchResultsViewController<SYSearchResult> = SYUISearchResultsTableViewController<SYSearchResult>()
+    
+    /// Search bar controller manages input field for search.
+    public var searchBarController = SYUISearchBarController()
+    
+    /// Delegate output for search controller.
+    public weak var delegate: SYMKSearchViewControllerDelegate?
+    
+    /// Search find results around this location. If it is not set, user location is used.
+    ///
+    /// If search location is not set and user doesn't have valid location, search engine try to search whole world for most relevant results.
+    public var searchLocation: SYGeoCoordinate? {
+        didSet {
+            model?.location = searchLocation
+        }
+    }
+    
+    /// Max number of results search returns.
+    public var maxResultsCount = SYMKSearchModel.maxResultsDefault {
+        didSet {
+            model?.maxResultsCount = maxResultsCount
+        }
+    }
+    
+    // MARK: - Private properties
+    
+    private var model: SYMKSearchModel?
+    
+    // MARK: - Public methods
+    
+    override func sygicSDKInitialized() {
+        model = SYMKSearchModel(maxResultsCount: maxResultsCount, location: searchLocation)
+        searchBarController.delegate = self
+        if !searchBarController.searchText.isEmpty {
+            search(for: searchBarController.searchText)
+        }
+        resultsViewController.interactionBlock = { [weak self] in
+            _ = self?.searchBarController.resignFirstResponder()
+        }
+        resultsViewController.selectionBlock = { [weak self] searchResult in
+            guard let weakSelf = self else { return }
```

Comment:
guard let 'self' = self else { return }

Reply:
https://benscheirman.com/2018/09/capturing-self-with-swift-4-2/ ... however this has been noted as a compiler bug and you should probably avoid using it, as it can change in a future release.